Mihir: Task 505-511: proto_qa & peixian/rtGender

Mda233 opened the pull request.

Mda233 commented:
I have verified my json files at my local end.

swarooprm commented:
@Mihir3009, Wondering if you can help review these tasks?

Mda233 commented:
Hi @swarooprm, can I keep working on creating more tasks and pull again?

Mda233 commented:
@swarooprm, I have an idea about the positive/negative response generation based on the post. Do you think if these are good tasks?

Mihir3009 commented:
1. Don't remove that Swaroop added from Readme.
2. Add a link to a dataset in the source. Something like this, "Source": [Dataset Name (Link - )]
3. Add "Contributors", "Categories" to all JSON files

Task 507:

1. Change definition like this:

`In this task, you are given a post. Your task is to identify the gender of the source poster or speaker based on the given post. Generate "W" if the gender is female, and generate "M" if the gender is male.`

2. Change category to "Answer Generation".

Task 508:

1. Change definition: Please make sure that how many sentiments are there, and explain each sentiment in the definition.
2. Make sure your data in Instances is almost balanced for each class not biased towards only one class
3. Change Category to "Classification"

Task 509:

1. Change category to "Classification"
2. Let's make the definition more comprehensive. Change it to this:

`In this task, you are given a source poster and a corresponding response from Facebook, TED, Fitocracy, and Reddit. Your task is to classify the given poster-response pair into four categories: 1) Content if it's the content only, 2) Poster, if it's the poster only, 3) ContentPoster if it's related to both, and 4) Irrelevant if it's not relevant at all.`

Task 510:

1. Change category to "Sentence Generation"
2. Change definition to this:

`In this task, you are given a poster from Facebook, TED, Fitocracy, and Reddit. Your task is to generate a response with positive sentiment for the given poster. `

Task 511:

1. Change category to "Sentence Generation"
2. Change definition to this:

`In this task, you are given a poster from Facebook, TED, Fitocracy, and Reddit. Your task is to generate a response with negative sentiment for the given poster. `

Update Readme summary:

Task 505: Given a question, generate a relevant answer to the question
Task 506: Given a group of answers, generate a question for all answers

Also, update categories accordingly.

swarooprm commented:
@Mihir3009, check if you are happy with the PR or have any other feedback?

Mihir3009 commented:
@swarooprm All comments have not been addressed yet.

Mda233 commented:
@Mihir3009 , All comments have been addressed, please check inside the files.

Mihir3009 commented:
I checked. You haven't updated this yet:

1. Don't remove that Swaroop added from Readme.
2. Add a link to a dataset in the source. Something like this, "Source": [Dataset Name (Link - )]
3. Add "Contributors", "Categories" to all JSON files

These are not addressed yet. Look at the files.

Mda233 commented:
1. Don't remove that Swaroop added from Readme.
 Do you mean this part?
Name | Summary | Category
-- | -- | --
task1001_squad1.1_question_generation | Generating guestions (based on SQuAD 1.1) | Question Generation
task1002_squad1.1_answer_generation | Generating answers to SQuAD 1.1 questions | Answer Generation

I added it back.

2. Add a link to a dataset in the source. Something like this, "Source": [Dataset Name (Link - )]
I truly did:
"Source": [
        "proto_qa (Link - https://huggingface.co/datasets/proto_qa)"
    ],
"Source": [
        "peixian/rtGender (Link - https://huggingface.co/datasets/peixian/rtGender)"
    ]

3. Add "Contributors", "Categories" to all JSON files
It's at the end of the JSON files like always.

Mda233 commented:
@Mihir3009 Please check the commit version called "Updated version." 
The check was failed due to a typo. So I committed again.

Mihir3009 commented:
Small Comments:

1. Add back Swaroop's JSON files also
2. Take contributors and categories in the beginning part, above definition, don't put it in the end.

Let me know, once you update that.

Mda233 commented:
@Mihir3009 , Please check. Thank you!
>>Add back Swaroop's JSON files also
Done.
>>Take contributors and categories in the beginning part, above definition, don't put it in the end.
I put it in the end because Swaroop's JSON files did so. I moved them to the beginning. 
>>Delete this '.DS_Store' file from the branch.
Deleted.

Mihir3009 commented:
@swarooprm I am happy with @Mda233's work. This PR looks good to me.

swarooprm commented:
Your HW2 is complete now.

Would you like to submit the same tasks to our big project repo https://github.com/allenai/natural-instructions-expansion/pulls? You just need to fork that repo, rename tasks starting from the latest number in the repo and create PR there. This way, you can be part of our research on "learning from instructions instead of learning from examples (conventional machine learning)". This is purely optional and your HW2 grade does not depend on this @Mda233

Mda233 commented:
Sure. I can do that.

On Fri, Sep 24, 2021 at 2:08 PM Swaroop Mishra ***@***.***>
wrote:

> Your HW2 is complete now.
>
> Would you like to submit the same tasks to our big project repo
> https://github.com/allenai/natural-instructions-expansion/pulls
> <https://urldefense.com/v3/__https://github.com/allenai/natural-instructions-expansion/pulls__;!!IKRxdwAv5BmarQ!PCnaiVw5s1yBPAu3xOVLAdVG569o63ng-VJLTMoBaTm5dbCwu5PYkbZyIAi9C2U$>?
> You just need to fork that repo, rename tasks starting from the latest
> number in the repo and create PR there. This way, you can be part of our
> research on "learning from instructions instead of learning from examples
> (conventional machine learning)". This is purely optional and your HW2
> grade does not depend on this @Mda233
> <https://urldefense.com/v3/__https://github.com/Mda233__;!!IKRxdwAv5BmarQ!PCnaiVw5s1yBPAu3xOVLAdVG569o63ng-VJLTMoBaTm5dbCwu5PYkbZyrCUf0n0$>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/swarooprm/CSE576HW/pull/40*issuecomment-926920169__;Iw!!IKRxdwAv5BmarQ!PCnaiVw5s1yBPAu3xOVLAdVG569o63ng-VJLTMoBaTm5dbCwu5PYkbZybRCyiug$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AVBCOSTW2U3V6SMNAA4BSLTUDTSDXANCNFSM5D7VHUOA__;!!IKRxdwAv5BmarQ!PCnaiVw5s1yBPAu3xOVLAdVG569o63ng-VJLTMoBaTm5dbCwu5PYkbZyt-OCkSk$>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://urldefense.com/v3/__https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675__;!!IKRxdwAv5BmarQ!PCnaiVw5s1yBPAu3xOVLAdVG569o63ng-VJLTMoBaTm5dbCwu5PYkbZyhOwGyxE$>
> or Android
> <https://urldefense.com/v3/__https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign*3Dnotification-email*26utm_medium*3Demail*26utm_source*3Dgithub__;JSUlJSU!!IKRxdwAv5BmarQ!PCnaiVw5s1yBPAu3xOVLAdVG569o63ng-VJLTMoBaTm5dbCwu5PYkbZyJF5Cg80$>.
>
>
-- 
DongAo Ma
School of Computing and Augmented Intelligence (SCAI)
Ira A. Fulton School of Engineering, Arizona State University
Email: ***@***.***

swarooprm commented:
@Mda233  Just a reminder that, if you plan to submit these tasks to the main Github repo and join our research project, Monday (27th Sept) is the deadline. Again, this is optional and independent of your HW2 grade.

Mda233 commented:
@swarooprm I am working on it. I Will finish before the ddl.

Mda233 commented:
@swarooprm Done!
